### PR TITLE
API Updates

### PR DESCRIPTION
**Pull request by richardm-stripe**
Codegen for openapi a8928d0.
r? @dcr-stripe
cc @stripe/api-libraries

## Changelog
* Add support for new resources `FundingInstructions` and `Terminal.Configuration`
* Add support for `CreateFundingInstructions` method on resource `Customer`
* Add support for `CustomerBalance` on `ChargePaymentMethodDetails`, `PaymentIntentConfirmPaymentMethodDataParams`, `PaymentIntentConfirmPaymentMethodOptionsParams`, `PaymentIntentPaymentMethodDataParams`, `PaymentIntentPaymentMethodOptionsParams`, `PaymentIntentPaymentMethodOptions`, `PaymentMethodParams`, and `PaymentMethod`
* Add support for `CashBalance` on `CustomerParams`
* Add support for `AmountDetails` on `PaymentIntent`
* Add support for `DisplayBankTransferInstructions` on `PaymentIntentNextAction`
* Add support for new value `customer_balance` on enum `PaymentMethodType`
* Add support for `ConfigurationOverrides` on `TerminalLocationParams` and `TerminalLocation`

### CODE REVIEW

**Review by kamil-stripe — APPROVED** [review]

File: example/generated_examples_test.go [hunk 1]
```diff
@@ -1606,3 +1607,52 @@ func TestTestHelpersTestClockAdvance(t *testing.T) {
 	result, _ := testhelpers_testclock.Advance("clock_xyz", params)
 	assert.NotNil(t, result)
 }
+
+func TestCustomerCreateFundingInstructions(t *testing.T) {
+	params := &stripe.CustomerCreateFundingInstructionsParams{
+		BankTransfer: &stripe.CustomerBankTransferParams{
```
Comment (dcr-stripe) [inline example/generated_examples_test.go:1613]: Need to fix this to include the custom method name (ie. it should be `CustomerCreateFundingInstructionsBankTransferParams`)

File: fundinginstructions.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	FundingInstructionsBankTransferFinancialAddressTypeZengin FundingInstructionsBankTransferFinancialAddressType = "zengin"
+)
+
+// The bank_transfer type
+type FundingInstructionsBankTransferType string
+
+// List of values that FundingInstructionsBankTransferType can take
+const (
+	FundingInstructionsBankTransferTypeEuBankTransfer FundingInstructionsBankTransferType = "eu_bank_transfer"
+	FundingInstructionsBankTransferTypeJpBankTransfer FundingInstructionsBankTransferType = "jp_bank_transfer"
```
Comment (dcr-stripe) [inline fundinginstructions.go:33]: EU and JP should be capitalized here.

File: paymentintent.go [hunk 3]
```diff
@@ -39,6 +39,32 @@ const (
...
+	PaymentIntentNextActionDisplayBankTransferInstructionsFinancialAddressTypeIban   PaymentIntentNextActionDisplayBankTransferInstructionsFinancialAddressType = "iban"
+	PaymentIntentNextActionDisplayBankTransferInstructionsFinancialAddressTypeZengin PaymentIntentNextActionDisplayBankTransferInstructionsFinancialAddressType = "zengin"
+)
+
+// Type of bank transfer
+type PaymentIntentNextActionDisplayBankTransferInstructionsType string
+
+// List of values that PaymentIntentNextActionDisplayBankTransferInstructionsType can take
+const (
+	PaymentIntentNextActionDisplayBankTransferInstructionsTypeJpBankTransfer PaymentIntentNextActionDisplayBankTransferInstructionsType = "jp_bank_transfer"
```
Comment (dcr-stripe) [inline paymentintent.go:65]: Same comment here for JP

File: paymentintent.go [hunk 4]
```diff
@@ -276,6 +302,55 @@ const (
...
+	PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferTypeIDBankTransfer PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferType = "id_bank_transfer"
+	PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferTypeJpBankAccount  PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferType = "jp_bank_account"
+	PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferTypeJpBankTransfer PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferType = "jp_bank_transfer"
+	PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferTypeMxBankAccount  PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferType = "mx_bank_account"
+	PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferTypeMxBankTransfer PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferType = "mx_bank_transfer"
+	PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferTypeUSBankAccount  PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferType = "us_bank_account"
+	PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferTypeUSBankTransfer PaymentIntentPaymentMethodOptionsCustomerBalanceBankTransferType = "us_bank_transfer"
```
Comment (dcr-stripe) [inline paymentintent.go:331]: We should fix the capitalizations here to be consistent

File: terminal_configuration.go [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+package stripe
+
+// An object containing device type specific settings for BBPOS WisePOS E readers
+type TerminalConfigurationBBPOSWisePOSEParams struct {
+	// A File ID representing an image you would like displayed on the reader.
+	Splashscreen *string `form:"splashscreen"`
+}
+
+// Tipping configuration for AUD
+type TerminalConfigurationTippingAudParams struct {
```
Comment (dcr-stripe) [inline terminal_configuration.go:16]: Need to fix these capitalizations to be consistent.

**Review by dcr-stripe — CHANGES_REQUESTED** [review]

**Review by kamil-stripe — APPROVED** [review]

**Review by dcr-stripe — APPROVED** [review]
LGTM

**Pull request merged**